source/extensions/quic_listeners/quiche/platform/quic_mem_slice_impl.cc

```diff
@@ -0,0 +1,51 @@
+// NOLINT(namespace-envoy)
+//
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "extensions/quic_listeners/quiche/platform/quic_mem_slice_impl.h"
+
+#include "common/common/assert.h"
+
+namespace quic {
+
+Envoy::Buffer::BufferFragmentImpl*
+QuicMemSliceImpl::allocateBufferAndFragment(QuicBufferAllocator* allocator, size_t length) {
+  auto bundle = reinterpret_cast<BufferFragmentBundle*>(
+      allocator->New(sizeof(BufferFragmentBundle) + length));
```

Comment:
Can you move the allocator for BufferFragmentBundle into the class itself? 2 ways to do it:
 - as an operator new override
 - as a static class method to allocate an instance.